core/src/test/scala/org/apache/spark/sql/delta/DeltaShowCreateTableSuite.scala

```diff
@@ -0,0 +1,194 @@
+    val table = "managed_table"
+    sql(s"CREATE TABLE `$table` (id LONG) USING delta")
+    val ddl = getShowCreateTable(table)
+    deleteTableAndData(table)
+    sql(ddl)
+    deleteTableAndData(table)
+  }
+
+  test(testName = "Test DDL Idempotency SHOW CREATE TABLE") {
+    val table = "managed_table"
+    sql(s"CREATE TABLE `$table` (id LONG) USING delta")
+    val ddl = getShowCreateTable(table)
+    deleteTableAndData(table)
+    sql(ddl)
+    val ddl1 = getShowCreateTable(table)
+    deleteTableAndData(table)
+    assert(ddl1.equals(ddl)) // table DDL are the same
+  }
+
+  test(testName = "Test DDL Comment SHOW CREATE TABLE") {
+    val table = "managed_table"
+    val comment = "This is a random comment on the table"
+    sql(s"CREATE TABLE `$table` (id LONG) USING delta COMMENT '$comment'")
+    val ddl = getShowCreateTable(table)
+    deleteTableAndData(table)
+    assert(ddl.contains(s"COMMENT '$comment'"))
+  }
+
+  test(testName = "Test DDL Partition SHOW CREATE TABLE") {
+    val table = "managed_table"
+    sql(
+      s"""CREATE TABLE $table (id INT, name STRING, age INT) USING delta
+         | PARTITIONED BY (age)""".stripMargin)
+    val ddl = getShowCreateTable(table)
+    assert(ddl.contains("PARTITIONED BY (age)"))
+    deleteTableAndData(table)
+  }
+
+  test(testName = "Test DDL Random Table Property SHOW CREATE TABLE") {
+    val table = "managed_table"
+    sql(
+      s"""CREATE TABLE $table (id INT, name STRING, age INT) USING delta
+         | PARTITIONED BY (age) TBLPROPERTIES
+         | ('foo'='bar','bar'='foo')""".stripMargin)
+    val ddl = getShowCreateTable(table)
+    deleteTableAndData(table)
+    assert(ddl.contains("PARTITIONED BY (age)"))
+    assert(ddl.contains("TBLPROPERTIES"))
+    assert(ddl.contains("'foo' = 'bar'"))
+    assert(ddl.contains("'bar' = 'foo'"))
+  }
+
+  test(testName = "Test DDL with full variations SHOW CREATE TABLE") {
+    withTempDir { foo =>
+      val table = "some_external_table"
+      val fooPath = foo.getCanonicalPath
+      sql(
+        s"""CREATE TABLE $table (id INT COMMENT "some id", name STRING, age INT) USING delta
+           | LOCATION "$fooPath"
+           | PARTITIONED BY (age)
+           | TBLPROPERTIES
+           | ('foo'='bar','bar'='foo')
+           | COMMENT "some comment"""".stripMargin)
+      val ddl = getShowCreateTable(table)
+      deleteTableAndData(table)
+      assert(ddl.contains("CREATE TABLE"))
+      assert(ddl.contains("`id` INT COMMENT 'some id'"))
+      assert(ddl.contains("'Type' = 'EXTERNAL'"))
+      assert(ddl.contains(s"LOCATION 'file:$fooPath'"))
+      assert(ddl.contains("USING delta"))
+      assert(ddl.contains("PARTITIONED BY (age)"))
+      assert(ddl.contains("TBLPROPERTIES"))
+      assert(ddl.contains("'foo' = 'bar'"))
+      assert(ddl.contains("'bar' = 'foo'"))
+    }
+  }
+
+//  test(testName = "Test DDL with Generated Column Expression SHOW CREATE TABLE") {
```

Comment:
Can we throw an unsupported error when seeing generated columns? Ignoring generated columns may surprise users.

Reply:
Ok, will make this change.

Reply:
@zsxwing this turns out to be a bit more complicated.

creating a generated table as follows:
```
val table = "people10m"
    io.delta.tables.DeltaTable.create(spark)
    .tableName(table)
      .addColumn("id", "INT")
      .addColumn("birthDate", "TIMESTAMP")
      .addColumn(io.delta.tables.DeltaTable.columnBuilder("c2")
                 .dataType(DateType)
                 .generatedAlwaysAs("CAST(birthDate AS DATE)")
                 .build())
    .execute()
```
The underlying column schema does not have the metadata for generatedAlwaysAs written to the map. This may be something as simple as a serialization issue - I will look into it further but it definitely appears to be a bug.

<img width="491" alt="Screen Shot 2022-07-07 at 11 39 14 PM" src="https://user-images.githubusercontent.com/4429967/177912269-4d417d7f-f311-4b21-80e3-5a90487e14eb.png">

Reply:
You would need to access DeltaLog.snapshot.metadata.schema in order to get the underlying map. We hide them from the user facing schema.